server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/remotestore/RemoteStoreStatsIT.java

```diff
@@ -0,0 +1,119 @@
+        Settings.Builder builder = Settings.builder()
+            .put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.SEGMENT)
+            .put(CLUSTER_REMOTE_STORE_ENABLED_SETTING.getKey(), true)
+            .put(CLUSTER_REMOTE_STORE_REPOSITORY_SETTING.getKey(), "my-segment-repo-1")
+            .put(CLUSTER_REMOTE_TRANSLOG_STORE_ENABLED_SETTING.getKey(), true)
+            .put(CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.getKey(), "my-translog-repo-1")
+            .put(settings);
+        return builder.build();
+    }
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder()
+            .put(super.featureFlagSettings())
+            .put(FeatureFlags.SEGMENT_REPLICATION_EXPERIMENTAL, "true")
+            .put(FeatureFlags.REMOTE_STORE, "true")
+            .build();
+    }
+
+    @Before
+    public void setup() {
+        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+        internalCluster().startClusterManagerOnlyNode();
+        Path absolutePath = randomRepoPath().toAbsolutePath();
+        assertAcked(
+            clusterAdmin().preparePutRepository("my-segment-repo-1")
+                .setType("fs")
+                .setSettings(Settings.builder().put("location", absolutePath))
+        );
+        assertAcked(
+            clusterAdmin().preparePutRepository("my-translog-repo-1")
+                .setType("fs")
+                .setSettings(Settings.builder().put("location", absolutePath))
+        );
+        assertAcked(
+            clusterAdmin().preparePutRepository("my-custom-repo")
+                .setType("fs")
+                .setSettings(Settings.builder().put("location", absolutePath))
+        );
+    }
+
+    public void testsSingleShardStatsResponse() throws Exception {
+        Settings settings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .build();
+        assertAcked(internalCluster().dataNodeClient().admin().indices().prepareCreate("test-idx-1").setSettings(settings).get());
+
+        ensureYellowAndNoInitializingShards("test-idx-1");
+        ensureGreen("test-idx-1");
+
+        indexData(10);
+
+        Thread.sleep(3000);
+        RemoteStoreStatsResponse remoteStoreStatsResponse = internalCluster().dataNodeClient()
+            .admin()
+            .cluster()
+            .prepareRemoteStoreStats("test-idx-1", "0")
+            .execute()
+            .actionGet();
+        RemoteStoreStats[] stats = remoteStoreStatsResponse.getShards();
+        assertEquals(remoteStoreStatsResponse.getTotalShards(), 0);
+        assertEquals(remoteStoreStatsResponse.getFailedShards(), 0);
+        assertEquals(remoteStoreStatsResponse.getSuccessfulShards(), 0);
+        assertEquals(stats.length, 0);
+        // assertEquals(stats[0].getStats().shardId.getIndexName(), "test-idx-1");
+        // assertEquals(stats[0].getStats().shardId.getId(), 0);
```

Comment:
we can remove this if we not required?

Reply:
yeah, ITs wont work until we have integration PR merged. i'll remove the entire file for now